times_reader/transforms.py

```diff
@@ -924,82 +974,28 @@ def remove_invalid_values(
 
 
 def process_units(
-    config: datatypes.Config,
-    tables: List[datatypes.EmbeddedXlTable],
-) -> List[datatypes.EmbeddedXlTable]:
-    commodity_units = set()
-    process_act_units = set()
-    process_cap_units = set()
-    currencies = set()
+    config: datatypes.Config, tables: Dict[str, DataFrame]
+) -> Dict[str, DataFrame]:
 
-    for table in tables:
-        if table.tag == datatypes.Tag.fi_comm:
-            commodity_units.update(table.dataframe["unit"].unique())
+    all_units = set()
 
-        if table.tag == datatypes.Tag.fi_process:
-            process_act_units.update(table.dataframe["tact"].unique())
-            process_cap_units.update(
-                [
-                    s.upper()
-                    for s in table.dataframe["tcap"].unique()
-                    if s != None and s != ""
-                ]
-            )
+    tags = {
+        datatypes.Tag.fi_comm: ["unit"],
+        datatypes.Tag.fi_process: ["tact", "tcap"],
+        datatypes.Tag.currencies: ["currency"],
+    }
 
-        if table.tag == datatypes.Tag.currencies:
-            currencies.update(table.dataframe["currency"].unique())
+    invalid_values = ["", None]
 
-    tables.append(
-        datatypes.EmbeddedXlTable(
-            tag="~UNITS_ACT",
```

Comment:
These other table tags are not required?

Reply:
No, the outputs that rely on them rely on more general tables now.